src/helpers/redirects.ts

```diff
@@ -174,4 +218,19 @@ export const generateRedirects = async ({
     to: HANDLER_FUNCTION_PATH,
     status: 200,
   })
+
+  const middlewareMatches = routesThatMatchMiddleware.size
+  if (middlewareMatches > 0) {
+    console.log(
+      yellowBright(outdent`
+        There ${
+          middlewareMatches === 1
+            ? `is one statically-generated or ISR route`
+            : `are ${middlewareMatches} statically-generated or ISR routes`
+        } that match a middleware function, which means they will always be served from the SSR function and will not use ISR or be served from the CDN.
+        If this was not intended, ensure that your middleware only matches routes that you intend to use SSR.
```

Comment:
[pebble] We'll likely want to include "match" in the conditional text, as we want the verb to match the noun's plurality (or lack thereof). I also edited a bit later to avoid some of the same potential awkwardness and improve clarity about what will always be served from SSR.

```suggestion
        There ${
          middlewareMatches === 1
            ? `is one statically-generated or ISR route that matches`
            : `are ${middlewareMatches} statically-generated or ISR routes that match`
        } a middleware function. Matched routes will always be served from the SSR function and will not use ISR or be served from the CDN.
        If this was not intended, ensure that your middleware only matches routes that you intend to use SSR.
```